# CrunchieMunchies

You work in marketing for a food company <b>myCorps</b>, which is developing a new kind of tasty, wholesome cereal called <b>CrunchieMunchies</b>. 

You want to demonstrate to consumers how healthy your cereal is in comparison to other leading brands, so you’ve dug up nutritional data on several different competitors.

Your task is to use <em>NumPy statistical calculations</em> to analyze this data and prove that your <b>CrunchieMunchies</b> is the healthiest choice for consumers.






# Task STEPS


1.First, import numpy.

In [1]:
# your code goes here
import numpy as np

2.Look over the <b><em>cereal.csv</em></b> file. This file contains the reported calorie amounts for different cereal brands. Load the data from the file and save it as <b><em>calorie_stats.</em></b>



In [2]:
# your code goes here
from numpy import genfromtxt
calorie_stats = genfromtxt('cereal.csv', delimiter=',')
print(calorie_stats)

[ 70. 120.  70.  50. 110. 110. 110. 130.  90.  90. 120. 110. 120. 110.
 110. 110. 100. 110. 110. 110. 100. 110. 100. 100. 110. 110. 100. 120.
 120. 110. 100. 110. 100. 110. 120. 120. 110. 110. 110. 140. 110. 100.
 110. 100. 150. 150. 160. 100. 120. 140.  90. 130. 120. 100.  50.  50.
 100. 100. 120. 100.  90. 110. 110.  80.  90.  90. 110. 110.  90. 110.
 140. 100. 110. 110. 100. 100. 110.]


3.There are <em>60 calories per serving of CrunchieMunchies</em>. How much <b>higher</b> is the <b>average calorie count</b> of your competition?

Save the answer to the variable <b>average_calories</b> and print the variable to the terminal to see the answer.


In [3]:
# your code goes here
average_calories=np.mean(calorie_stats)
average_calories

106.88311688311688

4.Does the <b>average calorie count</b> adequately reflect the distribution of the dataset? Let’s sort the data and see.

<b><em>Sort</em></b> the data and save the result to the variable <b>calorie_stats_sorted</b>. Print the sorted data to the terminal.


In [4]:
# your code goes here
calorie_stats_sorted=np.sort(calorie_stats)
calorie_stats_sorted

array([ 50.,  50.,  50.,  70.,  70.,  80.,  90.,  90.,  90.,  90.,  90.,
        90.,  90., 100., 100., 100., 100., 100., 100., 100., 100., 100.,
       100., 100., 100., 100., 100., 100., 100., 100., 110., 110., 110.,
       110., 110., 110., 110., 110., 110., 110., 110., 110., 110., 110.,
       110., 110., 110., 110., 110., 110., 110., 110., 110., 110., 110.,
       110., 110., 110., 110., 120., 120., 120., 120., 120., 120., 120.,
       120., 120., 120., 130., 130., 140., 140., 140., 150., 150., 160.])

5.Do you see what I’m seeing? Looks like <b><em>the majority of the cereals are higher than the mean</em></b>. Let’s see if the <b>median</b> is a better representative of the dataset.

Calculate the median of the dataset and save your answer to <b><em >median_calories</em></b>. Print the median so you can see how it compares to the mean.

In [5]:
# your code goes here
median_calories=np.median(calorie_stats)
median_calories


110.0

6.While the median demonstrates that <b><em><q>at least half of our values are over 100 calories</q></em></b>, it would be more impressive to show that a significant portion of the competition has a higher calorie count that CrunchieMunchies.

<b>Calculate different percentiles</b> and print them to the terminal until you find the lowest percentile that is greater than 60 calories. Save this value to the variable <b>nth_percentile</b>.


In [6]:
# your code goes here
print("90th percentile value = {}".format(np.percentile(calorie_stats, 90)))
print("80th percentile value = {}".format(np.percentile(calorie_stats, 80)))
print("70th percentile value = {}".format(np.percentile(calorie_stats, 70)))
print("60th percentile value = {}".format(np.percentile(calorie_stats, 60)))
print("50th percentile value = {}".format(np.percentile(calorie_stats, 50)))
print("40th percentile value = {}".format(np.percentile(calorie_stats, 40)))
print("30th percentile value = {}".format(np.percentile(calorie_stats, 30)))
print("20th percentile value = {}".format(np.percentile(calorie_stats, 20)))
print("10th percentile value = {}".format(np.percentile(calorie_stats, 10)))
print("5th percentile value = {}".format(np.percentile(calorie_stats, 5)))
print("2.5th percentile value = {}".format(np.percentile(calorie_stats, 2.5)))
print("4th percentile value = {}".format(np.percentile(calorie_stats, 4)))
print("3.5th percentile value = {}".format(np.percentile(calorie_stats, 3.5)))
print("3.25th percentile value = {}".format(np.percentile(calorie_stats, 3.25)))
print("3.37th percentile value = {}".format(np.percentile(calorie_stats, 3.37)))
print("3.31th percentile value = {}".format(np.percentile(calorie_stats, 3.31)))
print("3.28th percentile value = {}".format(np.percentile(calorie_stats, 3.28)))
print("3.30th percentile value = {}".format(np.percentile(calorie_stats, 3.30)))
print("3.29th percentile value = {}".format(np.percentile(calorie_stats, 3.29)))
nth_percentile=np.percentile(calorie_stats, 3.29)
print("nth percentile  =  {}".format(nth_percentile))

90th percentile value = 124.00000000000006
80th percentile value = 120.0
70th percentile value = 110.0
60th percentile value = 110.0
50th percentile value = 110.0
40th percentile value = 110.0
30th percentile value = 100.0
20th percentile value = 100.0
10th percentile value = 90.0
5th percentile value = 70.0
2.5th percentile value = 50.0
4th percentile value = 70.0
3.5th percentile value = 63.2
3.25th percentile value = 59.400000000000006
3.37th percentile value = 61.224
3.31th percentile value = 60.31199999999999
3.28th percentile value = 59.856
3.30th percentile value = 60.160000000000004
3.29th percentile value = 60.008
nth percentile  =  60.008


7.While the percentile shows us that<b><em><q>the majority of the competition has a much higher calorie count</q></em></b>, it’s an awkward concept to use in marketing materials.

Instead, let’s calculate the percentage of cereals that <b><em><q>have more than 60 calories per serving</q></em></b>. Save your answer to the variable <b><em>more_calories</em></b> and print it to the terminal

In [7]:
# your code goes here
greater_calories_sixty = calorie_stats[calorie_stats > 60]
more_calories=((greater_calories_sixty.shape[0]/calorie_stats.shape[0])*100)
print("percentage of having more than 60 calories per serving  =  {:.2f}".format(more_calories))

percentage of having more than 60 calories per serving  =  96.10


8.Wow! That’s a really high percentage. That’s going to be very useful when we promote CrunchieMunchies. But one question is, <b><em>how much variation exists in the dataset? </b></em></q>Can we make the generalization that most cereals have around 100 calories or is the spread even greater?

Calculate the amount of variation by finding the <b><em>standard deviation</em</b> Save your answer to <b><em>calorie_std</em></b> and print to the terminal. How can we incorporate this value into our analysis?

In [9]:
# your code goes here

calorie_stats_std=np.var(calorie_stats)
print("Standard deviation = {}".format(calorie_stats_std))

Standard deviation = 374.70062405127345


9.Write a short paragraph that sums up your findings and how you think this data could be used to 
<b>myCorp’s</b> advantage when marketing CrunchieMunchies.


# Conclusion

According to the finding as the most of the time cereal good at or greater than median=100. As given dataset having good quality calories in the cereals. Also there is greater dispersion in the dataset and only 3rd percentile are less than 60. In addition there is also good percentage 96.10 of more than 60 calories in the cereals. In data analysis some time values are misguiding then we use the median instead of mean. In this case, Mean and Median are nearly equal to each other and hence the less chance of greater outlier in this cereails. so we are recommended for marketing these cereals for CruncgieMunchies.  